### DISCUSSION TITLE
useQuery cancellation searching not work

### DISCUSSION THREAD

**Discussion by onlyhasbi** [TanStack/query#4916] [Q&A] [answered]
code example

[useQuery cancellation](https://codesandbox.io/s/react-query-cancellation-v3h5pu?file=/src/App.js)

*error showing on log and network request

thank you

**Comment by TkDodo** [comment, marked as answer]
`promise.cancel` is not supported in v4. It has been deprecated in v3. You have to use the AbortSignal provided by react-query, as shown in the docs:

https://tanstack.com/query/v4/docs/react/guides/query-cancellation#using-axios-v0220
